Code/GraphMol/Descriptors/AtomicEnvironmentVector.cpp

```diff
@@ -0,0 +1,845 @@
+//
+//  Copyright (C) 2020 Manan Goel
+//   @@ All Rights Reserved @@
+//  This file is part of the RDKit.
+//  The contents are covered by the terms of the BSD license
+//  which is included in the file license.txt, found at the root
+//  of the RDKit source tree.
+//
+
+#include <GraphMol/RDKitBase.h>
+#include <cmath>
+#include <Numerics/EigenSerializer/EigenSerializer.h>
+#include "AtomicEnvironmentVector.h"
+#include <Eigen/Dense>
+
+using namespace Eigen;
+
+namespace RDKit {
+namespace Descriptors {
+namespace ANI {
+
+//! Calculates the value a continuous smoothening function for a distance such
+//! that values
+// greater than the cutoff give 0
+/*!
+  \param distances A 2 dimensional array of pairwise distances
+  \param cutoff    A double value signifying cutoff distance
+
+  \return 2 dimensional array containing corresponding values computed by cutoff
+  function
+*/
+template <typename Derived>
+ArrayXXd CosineCutoff(ArrayBase<Derived> *distances, double cutoff) {
+  // Cosine cutoff function assuming all distances are less than the cutoff
+  PRECONDITION(cutoff > 0.0, "Cutoff must be greater than zero");
+  PRECONDITION(((*distances) <= cutoff).count() == distances->size(),
```

Comment:
Add a PRECONDITION for distances not being null
PRECONDITION(distances, "null distances passed in")